source/dub/project.d

```diff
@@ -294,7 +294,7 @@ class Project {
 			lbuildsettings.importPaths ~= NativePath(mainfil).parentPath.toNativeString;
 		bool firstTimePackage = true;
 		foreach (file; lbuildsettings.sourceFiles) {
-			if (file.endsWith(".d")) {
+			if (file.isSourceFile) {
```

Comment:
this should stay `.endsWith(".d")`, because this adds the source files / modules to the unittest module list. `.c` and `.i`files don't have any unittests / module name for this to be relevant.

Reply:
Reverted